src/bitcask.erl

```diff
@@ -404,11 +404,19 @@ fold(State, Fun, Acc0, MaxAge, MaxPut, SeeTombstonesP) ->
                     {ok, Files, FoldEpoch} ->
                         ExpiryTime = expiry_time(State#bc_state.opts),
                         SubFun = fun(K0,V,TStamp,{_FN,FTS,Offset,_Sz},Acc) ->
-                                         K = KT(K0),
-                                         case (TStamp < ExpiryTime) of
-                                             true ->
+                                         K = try
+                                                 KT(K0)
+                                             catch
+                                                 KeyTxErr ->
+                                                     {key_tx_error, {K0, KeyTxErr}}
+                                             end,
+                                         case {K, (TStamp < ExpiryTime)} of
+                                             {{key_tx_error, TxErr}, _} ->
+                                                 error_logger:error_msg("Error converting key ~p", [TxErr]),
```

Comment:
Wonder if this error would be better expressed as 

```erlang
  {{key_tx_error, {K0, TxErr}, _} ->
      error_logger:error_msg("Error converting key ~p because ~p", [K0, TxErr]),
```